### PR TITLE
Restrict all command input to ASCII

### LINKED ISSUE #147: Restrict input to ASCII characters app-wide
Currently, the validation checks for text are performed per-command in `ParserUtils.java`, and some commands admit non-ASCII text. Text validation should be done in the Command handler to only admit ASCII text.

### PR DESCRIPTION
**Pull request by conradsoon**
This fixes #147

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java
Comment (lordidiot) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:1]: Can we add a test with all printable characters to ensure that valid characters are allowed

Comment (conradsoon) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:1]: ok

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 1]
```diff
@@ -119,4 +120,27 @@ public void parseCommand_owe() throws Exception {
...
+    public void parseCommand_nonAsciiInput_throwsParseException() {
+        String nonAsciiInput = "add José";
+        assertThrows(ParseException.class,
+            AddressBookParser.MESSAGE_NON_ASCII, () -> parser.parseCommand(nonAsciiInput));
+    }
+
+    @Test
+    public void parseCommand_allAsciiInputAllowed() {
+        AddressBookParser parser = new AddressBookParser();
+        for (char c = '!'; c <= '~'; c++) {
```
Comment (lordidiot) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: Shouldn't this start with space?

```suggestion
        for (char c = ' '; c <= '~'; c++) {
```

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -34,7 +34,9 @@ public class AddressBookParser {
     /**
      * Used for initial separation of command word and args.
      */
+    public static final String MESSAGE_NON_ASCII = "Only printable ASCII characters are not allowed.";
```
Comment (lordidiot) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java]: Typo
```suggestion
    public static final String MESSAGE_NON_ASCII = "Only printable ASCII characters are allowed.";
```

**Review by lordidiot — CHANGES_REQUESTED** [review]
Some small changes then let's merge this

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java (hunk 1 above)
Comment (conradsoon) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: Space gets trimmed and gets a different error. Starting with '!' is correct.

Comment (conradsoon) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java]: Changed test case in #f7e1736

**Review by limjunxian1 — APPROVED** [review]
LGTM! thanks for the quick changes and writing a neater test case for this!

**Review by lordidiot — APPROVED** [review]
Thank you for working on this! This should help us handle an entire class of issues 😲

**Pull request merged**